experimental/IntersectionTheory/src/Main.jl

```diff
@@ -1839,32 +1839,13 @@ function intersection_matrix(a::Vector{}, b=nothing)
 end
 
 @doc raw"""
-     dual_basis(X::AbstractVariety, k::Int)
+    dual_basis(X::AbstractVariety)
 
-Compute the dual basis of the additive basis in codimension `k` given by
-`basis(X, k)` (the returned elements are therefore in codimension
-$\dim X-k$).
-"""
-function dual_basis(X::AbstractVariety, k::Int)
-  T = Dict{Int, Vector{elem_type(X.ring)}}
-  d = get_attribute!(X, :dual_basis) do
-    T()
-  end::T
-  if !(k in keys(d))
-    B = basis(X)
-    b_k = B[k+1]
-    b_comp = B[X.dim-k+1]
-    M = Matrix(inv(intersection_matrix(b_comp, b_k)))
-    d[k] = M * b_comp
-    d[X.dim-k] = transpose(M) * b_k
-  end
-  return d[k]
-end
+If `X` has been given a point class, return a `K`-basis of the Chow ring of `X` which is dual to `basis(X)` with respect to the `K`-bilinear form defined by `intersection_matrix(X)`. Here, `K = base(X)`.
 
-@doc raw"""
-    dual_basis(X::AbstractVariety)
-    dual_basis(X::AbstractVariety, k::Int)
-If `K = base(X)`, return a `K`-basis of the Chow ring of `X` which is dual to `basis(X)` with respect to the `K`-bilinear form defined by `intersection_matrix(X)` (return the elements of degree `k` in the dual basis).
+dual_basis(X::AbstractVariety, k::Int)
```

Comment:
```suggestion
    dual_basis(X::AbstractVariety, k::Int)
```
this should be indented by 4 spaces to be formatted as a signature

Reply:
Thx, done below!